event.go

```diff
@@ -239,39 +244,53 @@ func (e *Event) RawJSON(key string, b []byte) *Event {
 	return e
 }
 
-// AnErr adds the field key with err as a string to the *Event context.
+// AnErr adds the field key with serialized err to the *Event context.
 // If err is nil, no field is added.
 func (e *Event) AnErr(key string, err error) *Event {
+	marshaled := ErrorMarshalFunc(err)
+	switch m := marshaled.(type) {
+	case nil:
+		return e
+	case LogObjectMarshaler:
+		return e.Object(key, m)
+	case string:
+		return e.Str(key, m)
+	case error:
+		return e.Str(key, m.Error())
+	default:
+		return e.Interface(key, m)
+	}
+}
+// Errs adds the field key with errs as an array of serialized errors to the
+// *Event context.
+func (e *Event) Errs(key string, errs []error) *Event {
 	if e == nil {
 		return e
 	}
-	if err != nil {
-		e.buf = enc.AppendError(enc.AppendKey(e.buf, key), err)
-	}
-	return e
-}
 
-// Errs adds the field key with errs as an array of strings to the *Event context.
-// If err is nil, no field is added.
-func (e *Event) Errs(key string, errs []error) *Event {
-	if e == nil {
-		return e
+	arr := Arr()
+	for _, err := range errs {
+		marshaled := ErrorMarshalFunc(err)
+		switch m := marshaled.(type) {
+		case LogObjectMarshaler:
```

Comment:
Don't you miss the `nil` case here?

Reply:
Probably not, my reasoning is that nil will be handled by the default case in which it is serialized using the interface method (which will serialize it as "null").